Scripts/CI/README_Metadata_StyleCheck/metadata_style_checker.py

```diff
@@ -0,0 +1,338 @@
+import os
+import re
+import json
+import typing
+import argparse
+import glob
+from pathlib import Path
+
+# region Global sets
+# A set of category folder names in current sample viewer.
+categories = {
+    'analysis',
+    'display_information',
+    'editing',
+    'feature_layers',
+    'geometry',
+    'group_layers',
+    'hydrography',
+    'image_layers',
+    'kml',
+    'local_server',
+    'map',
+    'map_view',
+    'network_analysis',
+    'ogc',
+    'portal',
+    'raster',
+    'scene',
+    'search',
+    'symbology',
+    'tiled_layers',
+    'utility_network'
+}
+# endregion
+
+def sub_special_char(string: str) -> str:
+    """
+    Check and substitute if a string contains special characters.
+    :param string: The input string.
+    :return: A new string
+    """
+    # regex = re.compile('[@_!#$%^&*<>?|/\\}{~:]')
```

Comment:
Should we remove the commented out regex? It looks identical to the one below anyway?